src/ui/map.js

```diff
@@ -3268,7 +3280,19 @@ class Map extends Camera {
 
         if (this._loaded && !this._fullyLoaded && !somethingDirty) {
             this._fullyLoaded = true;
-            // Following line is billing related code. Do not change. See LICENSE.txt
+            // Following lines are billing and metrics related code. Do not change. See LICENSE.txt
+            if (this._enablePerformanceMetricsCollection) {
+                postPerformanceEvent(this._requestManager._customAccessToken, {
+                    width: this.painter.width,
+                    height: this.painter.height,
+                    interactionRange: this._interactionRange,
+                    terrain: this.painter.style.getTerrain(),
+                    fog: this.painter.style.getFog(),
```

Comment:
Should we cast it to booleans? Or is it intended so we can measure the different fog/terrain configurations?


```suggestion
                    terrain: !!this.painter.style.getTerrain(),
                    fog: !!this.painter.style.getFog(),
```

Reply:
Yes we can do that here, it's done further down this code path [here](https://github.com/mapbox/mapbox-gl-js/pull/12343/files#diff-ac38f489bbc239740d862fcaaec961fc80633f3e05ea2f9e415eb2577d2a541aR121), but it's better to constrain to a more focused type earlier like you're suggesting.

The intention for including these is to be able to bucket comparable maps together when inspecting performance traces based on a high level style specification modifiers. Since a map with terrain and/or fog might have different performance capabilities as it uses very different internal code paths (e.g. fog adds a few extra shader statement to  all the shaders, terrain goes through a different rendering technique with draping...), we want to separate performance comparison between them based on these entries.

Reply:
Updated in https://github.com/mapbox/mapbox-gl-js/commit/af215cf90ce754460ccfd84c86148f40b69569ce.